src/mcts/search.cc

```diff
@@ -1145,8 +1145,49 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
       }
       return NodeToProcess::Collision(node, depth, collision_limit);
     }
-    // Either terminal or unexamined leaf node -- the end of this playout.
-    if (node->IsTerminal() || !node->HasChildren()) {
+    // If terminal, we either found a twofold draw to be reverted, or
+    // reached the end of this playout.
+    if (node->IsTerminal()) {
+      // Probably best place to check for two-fold draws consistently.
+      // Depth starts with 1 at root, so real depth is depth - 1.
+      if (node->IsTwoFoldTerminal() && depth - 1 < node->GetM()) {
+        // Check whether first repetition was before root. If yes, remove
+        // terminal status of node and revert all visits in the tree.
```

Comment:
This comment really should be before the if.

Also probably worth mentioning that first repetition for a two fold terminal can only be before root under tree reuse from a previous move.

Reply:
Done.